### PR TITLE
Use git hash-object instead of git ls-files for inputs hashing

### PR DESCRIPTION
**Pull request by gsoltis**
Passing `"inputs"` directly to `git ls-files` uses `git`'s globbing, which is not the same as the rest of `turbo`. Do the globbing ourselves instead, and use `git hash-object`.

### CODE REVIEW

**Comment by gsoltis** [comment]
Fixes #1250

**Comment by gsoltis** [comment]
Fixes #1249

**Comment by gsoltis** [comment]
Marking as "on hold" only to prevent automerge

**Review by tknickman — APPROVED** [review]
Looks good to me.

**Review by mehulkar — APPROVED** [review]
Nice! I don't understand every line of Go, but I understand the gist of the work and it seems like good clean up!

File: cli/internal/context/context.go [hunk 1]
```diff
@@ -482,7 +483,7 @@ func calculateGlobalHash(rootpath fs.AbsolutePath, rootPackageJSON *fs.PackageJS
 		globalDepsPaths[i] = turbopath.AbsoluteSystemPathFromUpstream(path)
 	}
 
-	globalFileHashMap, err := fs.GetHashableDeps(rootpath, globalDepsPaths)
+	globalFileHashMap, err := hashing.GetHashableDeps(rootpath, globalDepsPaths)
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply cli/internal/context/context.go:486]: It's normal to infer it, but if need be you can prefix an import with a token, like [here](https://github.com/vercel/turborepo/blob/18c3ef6d93000b9a6210ebb4aef4ee670c962736/cli/internal/context/context.go#L21)

**Comment by nathanhammond** [comment]
We have two more places we need to clean up that are somewhat-related, I'll quickly pick those off and delete all of `scm`.

- https://github.com/vercel/turborepo/blob/c57f7d263afde5b394e436cdd325eed0815d3395/cli/internal/taskhash/taskhash.go#L99
- https://github.com/vercel/turborepo/blob/c57f7d263afde5b394e436cdd325eed0815d3395/cli/internal/scope/scope.go#L148

**Pull request merged**